src/librustc_parse/parser/item.rs

```diff
@@ -647,7 +678,8 @@ impl<'a> Parser<'a> {
 
     /// Parses associated items.
     fn parse_assoc_item(&mut self, req_name: ReqName) -> PResult<'a, Option<Option<P<AssocItem>>>> {
-        Ok(self.parse_item_(req_name)?.map(|Item { attrs, id, span, vis, ident, kind, tokens }| {
+        let item = self.parse_item_(req_name)?;
+        Ok(item.map(|Item { attrs, id, span, vis, ident, kind, tokens }| {
```

Comment:
This should handle the missing closing brace case somehow.

Reply:
Isn't that handled in `parse_item_list`? Not sure what the relation to this PR is though.